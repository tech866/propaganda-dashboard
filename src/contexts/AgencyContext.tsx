'use client';

import React, { createContext, useContext, useEffect, useState } from 'react';
import { useUser, useOrganization } from '@clerk/nextjs';
import { supabase } from '@/lib/supabase';
import { UserRole } from '@/lib/clerk';

// Agency data interface
export interface Agency {
  id: string;
  name: string;
  registration_date: string;
  subscription_plan: 'basic' | 'professional' | 'enterprise';
  contact_info?: {
    phone?: string;
    email?: string;
    website?: string;
  };
  billing_address?: string;
  active_status: boolean;
  created_at: string;
  updated_at: string;
}

// Agency context interface
interface AgencyContextType {
  agency: Agency | null;
  isLoading: boolean;
  error: string | null;
  refreshAgency: () => Promise<void>;
  canManageAgency: boolean;
  canViewFinancial: boolean;
  canManageUsers: boolean;
}

// Create the context
const AgencyContext = createContext<AgencyContextType | undefined>(undefined);

// Agency provider component
export function AgencyProvider({ children }: { children: React.ReactNode }) {
  // Temporarily disable Clerk hooks for development
  // TODO: Re-enable when proper Clerk keys are configured
  const [agency, setAgency] = useState<Agency | null>(null);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<string | null>(null);
  
  // Mock agency for development
  React.useEffect(() => {
    if (process.env.NODE_ENV === 'development' && process.env.NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY?.includes('placeholder')) {
      setAgency({
        id: 'dev-agency-1',
        name: 'Development Agency',
        registration_date: '2024-01-01',
        subscription_plan: 'professional',
        contact_info: {
          phone: '+1-555-0123',
          email: 'dev@agency.com',
          website: 'https://dev-agency.com'
        },
        billing_address: '123 Dev Street, Dev City, DC 12345',
        active_status: true,
        created_at: '2024-01-01T00:00:00Z',
        updated_at: '2024-01-01T00:00:00Z'
      });
      setIsLoading(false);
      return;
    }
  }, []);
  
  // Original Clerk implementation (commented out for development)
  // const { user, isLoaded: userLoaded } = useUser();
  // const { organization, isLoaded: orgLoaded } = useOrganization();

  // Fetch agency data from Supabase
  const fetchAgency = async (agencyId: string) => {
    try {
      setIsLoading(true);
      setError(null);

      const { data, error: fetchError } = await supabase
        .from('agencies')
        .select('*')
        .eq('id', agencyId)
        .eq('active_status', true)
        .single();

      if (fetchError) {
        throw fetchError;
      }

      setAgency(data);
    } catch (err) {
      console.error('Error fetching agency:', err);
      setError(err instanceof Error ? err.message : 'Failed to fetch agency data');
      setAgency(null);
    } finally {
      setIsLoading(false);
    }
  };

  // Refresh agency data
  const refreshAgency = async () => {
    if (organization?.id) {
      await fetchAgency(organization.id);
    }
  };

  // Original Clerk useEffect (commented out for development)
  // useEffect(() => {
  //   if (!userLoaded || !orgLoaded) {
  //     setIsLoading(true);
  //     return;
  //   }

  //   if (!user || !organization) {
  //     setAgency(null);
  //     setIsLoading(false);
  //     return;
  //   }

  //   // Fetch agency data from Supabase using organization ID
  //   fetchAgency(organization.id);
  // }, [user, userLoaded, organization, orgLoaded]);

  // Mock permission checks for development
  const canManageAgency = true; // Admin user in development
  const canViewFinancial = true; // Admin user in development
  const canManageUsers = true; // Admin user in development

  const value: AgencyContextType = {
    agency,
    isLoading,
    error,
    refreshAgency,
    canManageAgency,
    canViewFinancial,
    canManageUsers,
  };

  return (
    <AgencyContext.Provider value={value}>
      {children}
    </AgencyContext.Provider>
  );
}

// Custom hook to use the agency context
export function useAgency() {
  const context = useContext(AgencyContext);
  if (context === undefined) {
    throw new Error('useAgency must be used within an AgencyProvider');
  }
  return context;
}

// Higher-order component for agency-based access control
export function withAgency<T extends object>(
  Component: React.ComponentType<T>,
  requiredPermission?: 'manage' | 'financial' | 'users'
) {
  return function AgencyProtectedComponent(props: T) {
    const { agency, isLoading, canManageAgency, canViewFinancial, canManageUsers } = useAgency();

    if (isLoading) {
      return (
        <div className="flex items-center justify-center p-4">
          <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-blue-600"></div>
        </div>
      );
    }

    if (!agency) {
      return (
        <div className="flex items-center justify-center p-8">
          <div className="text-center">
            <h3 className="text-lg font-semibold text-gray-900 mb-2">No Agency Access</h3>
            <p className="text-gray-600">
              You need to be part of an agency to access this content.
            </p>
          </div>
        </div>
      );
    }

    // Check specific permissions if required
    if (requiredPermission === 'manage' && !canManageAgency) {
      return (
        <div className="flex items-center justify-center p-8">
          <div className="text-center">
            <h3 className="text-lg font-semibold text-gray-900 mb-2">Access Denied</h3>
            <p className="text-gray-600">
              You don't have permission to manage this agency.
            </p>
          </div>
        </div>
      );
    }

    if (requiredPermission === 'financial' && !canViewFinancial) {
      return (
        <div className="flex items-center justify-center p-8">
          <div className="text-center">
            <h3 className="text-lg font-semibold text-gray-900 mb-2">Access Denied</h3>
            <p className="text-gray-600">
              You don't have permission to view financial data.
            </p>
          </div>
        </div>
      );
    }

    if (requiredPermission === 'users' && !canManageUsers) {
      return (
        <div className="flex items-center justify-center p-8">
          <div className="text-center">
            <h3 className="text-lg font-semibold text-gray-900 mb-2">Access Denied</h3>
            <p className="text-gray-600">
              You don't have permission to manage users.
            </p>
          </div>
        </div>
      );
    }

    return <Component {...props} />;
  };
}

// Component for conditional rendering based on agency permissions
export function AgencyGuard({ 
  permission, 
  children, 
  fallback 
}: { 
  permission?: 'manage' | 'financial' | 'users';
  children: React.ReactNode;
  fallback?: React.ReactNode;
}) {
  const { agency, isLoading, canManageAgency, canViewFinancial, canManageUsers } = useAgency();

  if (isLoading) {
    return (
      <div className="flex items-center justify-center p-4">
        <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-blue-600"></div>
      </div>
    );
  }

  if (!agency) {
    return fallback || null;
  }

  // Check specific permissions if required
  if (permission === 'manage' && !canManageAgency) {
    return fallback || null;
  }

  if (permission === 'financial' && !canViewFinancial) {
    return fallback || null;
  }

  if (permission === 'users' && !canManageUsers) {
    return fallback || null;
  }

  return <>{children}</>;
}

export default AgencyContext;
