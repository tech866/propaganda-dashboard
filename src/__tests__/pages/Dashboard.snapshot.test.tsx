import React from 'react';
import { render } from '@testing-library/react';
import Dashboard from '@/app/dashboard/page';
import { useRole } from '@/contexts/RoleContext';
import { useAgency } from '@/contexts/AgencyContext';

// Mock the contexts
jest.mock('@/contexts/RoleContext', () => ({
  useRole: jest.fn()
}));

jest.mock('@/contexts/AgencyContext', () => ({
  useAgency: jest.fn()
}));

// Mock child components
jest.mock('@/components/layout/ModernDashboardLayout', () => {
  return function MockModernDashboardLayout({ children }: { children: React.ReactNode }) {
    return <div data-testid="dashboard-layout">{children}</div>;
  };
});

jest.mock('@/components/dashboard/EnhancedDashboard', () => {
  return function MockEnhancedDashboard() {
    return <div data-testid="enhanced-dashboard">Enhanced Dashboard</div>;
  };
});

const mockUseRole = useRole as jest.MockedFunction<typeof useRole>;
const mockUseAgency = useAgency as jest.MockedFunction<typeof useAgency>;

describe('Dashboard Page Snapshot', () => {
  beforeEach(() => {
    mockUseAgency.mockReturnValue({
      agency: {
        id: 'agency-1',
        name: 'Test Agency',
        registration_date: '2024-01-01',
        subscription_plan: 'professional',
        contact_info: {
          phone: '+1-555-0123',
          email: 'test@agency.com',
          website: 'https://test-agency.com'
        },
        billing_address: '123 Test Street, Test City, TC 12345',
        active_status: true,
        created_at: '2024-01-01T00:00:00Z',
        updated_at: '2024-01-01T00:00:00Z'
      },
      isLoading: false,
      error: null
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot for authenticated admin user', () => {
    const mockUser = {
      id: 'user-1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'admin'
    };

    mockUseRole.mockReturnValue({
      user: mockUser,
      isLoading: false,
      hasAnyRole: jest.fn().mockReturnValue(true),
      hasPermission: jest.fn().mockReturnValue(true),
    });

    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for loading state', () => {
    mockUseRole.mockReturnValue({
      user: null,
      isLoading: true,
      hasAnyRole: jest.fn(),
      hasPermission: jest.fn(),
    });

    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for unauthenticated user', () => {
    mockUseRole.mockReturnValue({
      user: null,
      isLoading: false,
      hasAnyRole: jest.fn(),
      hasPermission: jest.fn(),
    });

    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for unauthorized user', () => {
    const mockUser = {
      id: 'user-1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'client_user'
    };

    mockUseRole.mockReturnValue({
      user: mockUser,
      isLoading: false,
      hasAnyRole: jest.fn().mockReturnValue(false),
      hasPermission: jest.fn().mockReturnValue(false),
    });

    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot for CEO user', () => {
    const mockUser = {
      id: 'user-1',
      name: 'Jane Smith',
      email: 'jane@example.com',
      role: 'ceo'
    };

    mockUseRole.mockReturnValue({
      user: mockUser,
      isLoading: false,
      hasAnyRole: jest.fn().mockReturnValue(true),
      hasPermission: jest.fn().mockReturnValue(true),
    });

    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
