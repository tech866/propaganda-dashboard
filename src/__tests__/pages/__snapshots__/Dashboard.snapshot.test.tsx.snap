// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dashboard Page Snapshot matches snapshot for CEO user 1`] = `
<div
  data-testid="dashboard-layout"
>
  <div
    class="p-6 space-y-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h1
        class="text-3xl font-bold text-white"
      >
        Dashboard
      </h1>
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-700 disabled:bg-blue-800 disabled:opacity-50 text-white rounded-lg transition-colors"
        disabled=""
      >
        Loading...
      </button>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Key Performance Indicators
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
        >
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Client Performance
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Recent Campaigns
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Agency Overview
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-4"
        >
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="fixed bottom-4 right-4 bg-slate-800 border border-slate-700 rounded-lg p-3 shadow-lg"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"
        />
        <span
          class="text-sm text-white"
        >
          Loading dashboard data...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page Snapshot matches snapshot for authenticated admin user 1`] = `
<div
  data-testid="dashboard-layout"
>
  <div
    class="p-6 space-y-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h1
        class="text-3xl font-bold text-white"
      >
        Dashboard
      </h1>
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-700 disabled:bg-blue-800 disabled:opacity-50 text-white rounded-lg transition-colors"
        disabled=""
      >
        Loading...
      </button>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Key Performance Indicators
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
        >
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Client Performance
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Recent Campaigns
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Agency Overview
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-4"
        >
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="fixed bottom-4 right-4 bg-slate-800 border border-slate-700 rounded-lg p-3 shadow-lg"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"
        />
        <span
          class="text-sm text-white"
        >
          Loading dashboard data...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page Snapshot matches snapshot for loading state 1`] = `
<div
  data-testid="dashboard-layout"
>
  <div
    class="p-6 space-y-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h1
        class="text-3xl font-bold text-white"
      >
        Dashboard
      </h1>
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-700 disabled:bg-blue-800 disabled:opacity-50 text-white rounded-lg transition-colors"
        disabled=""
      >
        Loading...
      </button>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Key Performance Indicators
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
        >
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Client Performance
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Recent Campaigns
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Agency Overview
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-4"
        >
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="fixed bottom-4 right-4 bg-slate-800 border border-slate-700 rounded-lg p-3 shadow-lg"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"
        />
        <span
          class="text-sm text-white"
        >
          Loading dashboard data...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page Snapshot matches snapshot for unauthenticated user 1`] = `
<div
  data-testid="dashboard-layout"
>
  <div
    class="p-6 space-y-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h1
        class="text-3xl font-bold text-white"
      >
        Dashboard
      </h1>
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-700 disabled:bg-blue-800 disabled:opacity-50 text-white rounded-lg transition-colors"
        disabled=""
      >
        Loading...
      </button>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Key Performance Indicators
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
        >
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Client Performance
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Recent Campaigns
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Agency Overview
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-4"
        >
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="fixed bottom-4 right-4 bg-slate-800 border border-slate-700 rounded-lg p-3 shadow-lg"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"
        />
        <span
          class="text-sm text-white"
        >
          Loading dashboard data...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page Snapshot matches snapshot for unauthorized user 1`] = `
<div
  data-testid="dashboard-layout"
>
  <div
    class="p-6 space-y-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h1
        class="text-3xl font-bold text-white"
      >
        Dashboard
      </h1>
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-700 disabled:bg-blue-800 disabled:opacity-50 text-white rounded-lg transition-colors"
        disabled=""
      >
        Loading...
      </button>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Key Performance Indicators
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
        >
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
          <div
            class="bg-slate-700 h-24 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Client Performance
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Recent Campaigns
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="space-y-3"
        >
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
          <div
            class="bg-slate-700 h-16 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="bg-slate-800/50 backdrop-blur-sm border border-slate-700 rounded-2xl p-6"
    >
      <h2
        class="text-xl font-semibold text-white mb-4"
      >
        Agency Overview
      </h2>
      <div
        class="animate-pulse"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-4"
        >
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
          <div
            class="bg-slate-700 h-20 rounded-lg"
          />
        </div>
      </div>
    </div>
    <div
      class="fixed bottom-4 right-4 bg-slate-800 border border-slate-700 rounded-lg p-3 shadow-lg"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="animate-spin rounded-full h-4 w-4 border-b-2 border-blue-600"
        />
        <span
          class="text-sm text-white"
        >
          Loading dashboard data...
        </span>
      </div>
    </div>
  </div>
</div>
`;
